src/SBVRType.d.ts

```diff
@@ -0,0 +1,47 @@
+interface SBVRType<I,O> {
+  types: ConcreteTypes
+
+  validate(value: any, required: boolean, cb:Callback<I>): void
+
+  fetchProcessing?(data:I, cb:Callback<O>): void
+
+  nativeProperties?: NativeProperties
+
+  nativeFactTypes?: NativeFactTypes
+}
+
+type Callback<T> = (err?: any, data?: T) => void
+
+declare class ConcreteTypes {
+  postgres: DBTypeBuilder
+  mysql: DBTypeBuilder
+  websql: DBTypeBuilder
+  odata: ODataTypeBuilder
+}
+
+type DBTypeBuilder =
+    string
+  | ((necessity: string, index: string) => string)
+
+declare class ODataTypeBuilder {
+  name: string
+  complexType?: string
+}
+
+// It would be nice to be a bit smarter about these verb and term types. Unfortunately indexed types can only be string or number atm and this excludes evern type aliases or a sum of possible string literals
+type Verb = string
+type Term = string
+
+type AbstractSQL = (string | string[])[]
```

Comment:
We could keep this as `any` for the time being (as Page mentioned in another thread) eventually this should become some type defined in `abstract-sql-compiler`